crates/ruff_python_formatter/src/statement/stmt_try.rs

```diff
@@ -1,13 +1,175 @@
+    ) -> FormatResult<()> {
+        let ExceptHandler::ExceptHandler(except_handler_except_handler) = item;
+        let ExceptHandlerExceptHandler {
+            range: _,
+            type_,
+            name,
+            body,
+        } = except_handler_except_handler;
+
+        let comments_info = f.context().comments().clone();
+        let leading_comments =
+            comments_info.leading_comments(except_handler_except_handler.as_any_node_ref());
+        if let [first_leading_comment, ..] = leading_comments {
+            if lines_before(
+                first_leading_comment.slice().start(),
+                f.context().contents(),
+            ) > 1
+            {
+                write!(f, [empty_line()])?;
+            }
+        }
+
+        leading_node_comments(except_handler_except_handler).fmt(f)?;
+
+        write!(f, [text("except")])?;
+
+        if let Some(type_) = type_ {
+            write!(
+                f,
+                [space(), type_.format().with_options(Parenthesize::IfBreaks)]
+            )?;
+            if let Some(name) = name {
+                write!(
+                    f,
+                    [
+                        space(),
+                        text("as"),
+                        space(),
+                        dynamic_text(name.as_str(), None)
+                    ]
+                )?;
+            }
+        }
+        write!(f, [text(":"), block_indent(&body.format())])?;
+
+        dangling_node_comments(except_handler_except_handler).fmt(f)?;
+        trailing_node_comments(except_handler_except_handler).fmt(f)
+    }
+}
+
+impl<'ast> AsFormat<PyFormatContext<'ast>> for ExceptHandler {
+    type Format<'a> = FormatRefWithRule<
+        'a,
+        ExceptHandler,
+        FormatExceptHandler,
+        PyFormatContext<'ast>,
+    > where Self: 'a;
+
+    fn format(&self) -> Self::Format<'_> {
+        FormatRefWithRule::new(self, FormatExceptHandler::default())
+    }
+}
+
 impl FormatNodeRule<StmtTry> for FormatStmtTry {
     fn fmt_fields(&self, item: &StmtTry, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtTry {
+            range: _,
+            body,
+            handlers,
+            orelse,
+            finalbody,
+        } = item;
+
+        let comments_info = f.context().comments().clone();
+        let mut dangling_comments = comments_info.dangling_comments(item.as_any_node_ref());
+
+        write!(
+            f,
+            [
+                text("try:"),
+                hard_line_break(),
+                block_indent(&body.format()),
+            ]
+        )?;
+
+        for handler in handlers {
+            let handler_comments_start =
+                dangling_comments.partition_point(|comment| comment.slice().end() <= handler.end());
+            let (handler_comments, rest) = dangling_comments.split_at(handler_comments_start);
+            dangling_comments = rest;
```

Comment:
nit: can you just directly reassign `dangling_comments`?

Reply:
This may not work because `danglign_comments` isn't local to a single iteration of the for body. It must be reassigned for the next iteration.